worlds/pokemon_crystal/rom.py

```diff
@@ -0,0 +1,425 @@
+        player_text = convert_to_ingame_text(text[0])[:16]
+        # pad with terminator byte to keep alignment
+        player_text += [0x50] * (17 - len(player_text))
+        item_text = convert_to_ingame_text(text[1])[:16]
+        item_text.append(0x50)
+        # bank 1
+        bank = 0x75
+        table_offset_adr = item_name_table_adr + i * 5
+
+        if i >= item_name_bank1_capacity + item_name_bank2_capacity:
+            # if we somehow run out of capacity in both banks, just finish the table and break,
+            # there is a fallback string in the ROM, so it should handle this gracefully.
+            write_bytes(patch, [0xFF], item_name_table_adr + table_offset_adr)
+            print("oopsie")
+            break
+        if i + 1 < item_name_bank1_capacity:
+            text_offset = i * 34
+            text_adr = item_name_bank1 + text_offset
+        else:
+            # bank 2
+            bank = 0x76
+            text_offset = (i + 1 - item_name_bank1_capacity) * 34
+            text_adr = item_name_bank2 + text_offset
+        write_bytes(patch, player_text + item_text, text_adr)
+        # get the address within the rom bank (0x4000 - 0x7FFF)
+        text_bank_adr = (text_adr % 0x4000) + 0x4000
+        write_bytes(patch, text[2].to_bytes(2, "big"), table_offset_adr)
+        write_bytes(patch, text_bank_adr.to_bytes(2, "little"), table_offset_adr + 2)
+        write_bytes(patch, [bank], table_offset_adr + 4)
+    write_bytes(patch, [0xFF], item_name_table_adr + item_name_table_length - 1)
+
+    if world.options.randomize_static_pokemon:
+        for _static_name, pkmn_data in world.generated_static.items():
+            pokemon_id = data.pokemon[pkmn_data.pokemon].id
+            for address in pkmn_data.addresses:
+                cur_address = data.rom_addresses[address] + 1
+                write_bytes(patch, [pokemon_id], cur_address)
+
+    if world.options.randomize_starters:
+        for j, pokemon in enumerate(["CYNDAQUIL_", "TOTODILE_", "CHIKORITA_"]):
+            pokemon_id = data.pokemon[world.generated_starters[j][0]].id
+            for i in range(1, 5):
+                cur_address = data.rom_addresses["AP_Starter_" + pokemon + str(i)] + 1
+                write_bytes(patch, [pokemon_id], cur_address)
+                if i == 4:
+                    helditem = item_const_name_to_id(world.generated_starter_helditems[j])
+                    write_bytes(patch, [helditem], cur_address + 2)
+
+    if world.options.randomize_wilds:
+        for grass_name, grass_encounters in world.generated_wild.grass.items():
+            cur_address = data.rom_addresses["AP_WildGrass_" + grass_name] + 3
+            for i in range(3):  # morn, day, nite
+                for encounter in grass_encounters:
+                    pokemon_id = data.pokemon[encounter.pokemon].id
+                    write_bytes(patch, [encounter.level, pokemon_id], cur_address)
+                    cur_address += 2
```

Comment:
I don't think this `for i in range(3)` accomplishes anything; `i` is unused in here.